File: golang-code/ex7/escola.go
package main

type Escola struct {
	Nome        string
	Professores map[*Professor]bool
}

func (e *Escola) AdicionarProfessor(p *Professor) {
	if e.Professores == nil {
		e.Professores = make(map[*Professor]bool)
	}

	if _, exists := e.Professores[p]; !exists {
		e.Professores[p] = true
	}
}

File: python-code/ex7/escola.py
from professor import Professor

class Escola:
  def __init__(self, nome):
    self.nome = nome
    self.professores = []

  def adicionar_professor(self, professor):
    if professor not in self.professores:
      self.professores.append(professor)

  def __str__(self):
    return self.nome